test/app-luatest/http_client_test.lua

```diff
@@ -215,65 +242,69 @@ local function test_post_and_get(test, url, opts)
-    test:is(r.body, "hello world", "GET: default right body")
+    t.assert_equals(r.status, 200, "GET: default http code page exists")
+    t.assert_equals(r.body, "hello world", "GET: default right body")
 
     r = responses.get2
-    test:is(r.status, 200, "GET: http code page exists")
-    test:is(r.body, "abc", "GET: right body")
+    t.assert_equals(r.status, 200, "GET: http code page exists")
+    t.assert_equals(r.body, "abc", "GET: right body")
 
     r = responses.absent_get
-    test:is(r.status, 500, "GET: absent method http code page exists")
-    test:is(r.reason, 'Unknown', '500 - Unknown')
-    test:is(r.body, "No such method", "GET: absent method right body")
+    t.assert_equals(r.status, 500, "GET: absent method http code page exists")
+    t.assert_equals(r.reason, 'Unknown', '500 - Unknown')
+    t.assert_equals(r.body, "No such method", "GET: absent method right body")
 
     r = responses.empty_post
-    test:is(r.status, 200, "POST: good status")
-    test:ok(r.headers['header1'] == headers.header1 and
-        r.headers['header2'] == headers.header2, "POST: good headers")
-    test:isnil(r.body, "POST: empty body")
+    t.assert_equals(r.status, 200, "POST: good status")
+    t.assert(r.headers['header1'] == headers.header1 and
+             r.headers['header2'] == headers.header2, "POST: good headers")
+    t.assert_equals(r.body, nil, "POST: empty body")
 
     r = responses.good_post
-    test:is(r.status, 200, "POST: good status")
-    test:ok(r.headers['header1'] == headers.header1 and
-        r.headers['header2'] == headers.header2, "POST: good headers")
-    test:is(r.body, json_body, "POST: body")
+    t.assert_equals(r.status, 200, "POST: good status")
+    t.assert_equals(r.headers['header1'], headers.header1,
+                    "POST: good header 1")
+    t.assert_equals(r.headers['header2'], headers.header2,
+                    "POST: good header 2")
+    t.assert_equals(r.body, json_body, "POST: body")
 
     r = responses.good_put
-    test:is(r.status, 200, "PUT: good status")
-    test:ok(r.headers['header'] == headers.header and
-        r.headers['header2'] == headers.header2, "PUT: good headers")
+    t.assert_equals(r.status, 200, "PUT: good status")
+    t.assert_equals(r.headers['header'], headers.header,
+                    "PUT: good header 1")
+    t.assert_equals(r.headers['header2'], headers.header2,
+                    "PUT: good header 2")
 
     r = responses.bad_get
-    test:is(r.status, 404, "GET: http page not exists")
-    test:is(r.reason, 'Unknown', '404 - Unknown')
-    test:isnt(r.body:len(), 0, "GET: not empty body page not exists")
-    test:ok(string.find(r.body, "Not Found"),
-                "GET: right body page not exists")
+    t.assert_equals(r.status, 404, "GET: http page not exists")
+    t.assert_equals(r.reason, 'Unknown', '404 - Unknown')
+    t.assert(r.body:len() ~= 0, "GET: not empty body page not exists")
```

Comment:
It is better to use `t.assert_not_equals()`. In case of a failure, it'll show the bad value.